### ISSUE TITLE
etcd: allow configurable storage size

### ISSUE THREAD

**Issue by wurbanski** [kubermatic/kubermatic#11431]
### Description of the feature you would like to add / User story
etcd doesn't automatically grow it's database to the size on underlying PVC, it has a quota which is by default set to 2GiB. To override it, we need to add a flag `--quota-backend-bytes=<size>` to etcd's runtime command, which is currently not possible. Right now, resizing the PVC doesn't automatically imply that etcd has bigger storage available.

reference: https://etcd.io/docs/v3.5/dev-guide/limit/

### Solution details
etcd's storage scales with the PVC size up to a limit.

### Alternative approaches
Allow free-form customization of etcd's runtime command.

### Use cases
Reported by a customer, might be required in really high traffic clusters.

### Additional information
n/a

**Comment by csengerszabo** [comment]
Refinement outputs:

- This can be changed even after the cluster has been created, a flag should be added, where the customized storage size can be set up, and then it overrides the default.
- 8GB is the recommended maximum size, 2GB is the default. Any customization up to 8GB should be okay.
- Cluster should be paused while overriding.
- Small task, no objections or difficulties has come up

**Comment by csengerszabo** [comment]
Refinement outputs:

- This can be changed even after the cluster has been created, a flag should be added, where the customized storage size can be set up, and then it overrides the default.
- 8GB is the recommended maximum size, 2GB is the default. Any customization up to 8GB should be okay.
- Cluster should be paused while overriding.
- Small task, no objections or difficulties has come up

**Comment by ahmedwaleedmalik** [comment]
We won't continue with this option as discussed with the customer.

**Comment by ewassef** [comment]
@ahmedwaleedmalik can we at least configure the etcdlauncher to set the value to the pvc size? @JulianHansert ?

**Comment by ahmedwaleedmalik** [comment]
> etcdlauncher

The PVC size is already configurable https://github.com/kubermatic/kubermatic/blob/release/v2.22/pkg/apis/kubermatic/v1/cluster.go#L734

You can configure that at the seed level as well https://docs.kubermatic.com/kubermatic/v2.22/tutorials-howtos/project-and-cluster-management/seed-cluster/#:~:text=etcd%3A%0A%20%20%20%20%20%20clusterSize%3A%203-,diskSize,-%3A%205Gi.

**Comment by ewassef** [comment]
Yes, thats correct @ahmedwaleedmalik , but it doesnt get propagated to etcd so it will only use the default 2GB limit https://github.com/kubermatic/kubermatic/blob/main/cmd/etcd-launcher/main.go#L465 

We need to set the flag in the cmd https://etcd.io/docs/v3.5/op-guide/configuration/#command-line-flags:~:text=Maximum%20number%20of%20wal%20files%20to%20retain%20(0%20is%20unlimited).%0A%2D%2Dquota%2Dbackend%2Dbytes%20%270%27 

You can verify this by running `etcdctl endpoint status -w json` and you will see the max any etcd will be 2GB https://etcd.io/docs/v3.5/dev-guide/limit/#storage-size-limit

We ran into this and had to log into each pod and do the following:

`etcdctl compaction <lastrevision>; etcdctl defrag; etcdctl alarm disarm;` > this would be nice to put in that defragger cronjob

**Comment by csengerszabo** [comment]
Internal reference: 7806
